query-advisor/from-insight-to-conclusion.mdx

```diff
@@ -0,0 +1,199 @@
+
+![Workbooks baseline plan showing initial query performance before optimization](workbooks-baseline-plan.png)
+
+### Complete testing workflow
+The systematic approach to testing Query Advisor optimizations includes:
+
+1. **Baseline recording** - Capture current performance as reference point
+2. **Apply insights** - Use suggested rewrites to create variants
+3. **Benchmark results** - Compare performance improvements systematically
+4. **Validate changes** - Ensure optimizations work across parameter sets
+5. **Apply to production** - Implement verified improvements with confidence
+
+## Step 4: Apply Query Advisor insight
+
+In the workbook variant editor, Query Advisor provided an **"Apply"** button for the detected insight:
+
+![Workbooks interface showing Query Advisor sidebar with apply insight functionality](workbooks-query-advisor-sidebar.png)
+
+![Workbooks variant editor for applying Query Advisor insights to create optimized query versions](workbooks-query-advisor-apply-insight.png)
+
+### Automatic rewrite
+Query Advisor automatically modified the query to force better index selection:
+
+<CodeBlock language="sql">
+{`SELECT databases.*
+FROM databases
+WHERE databases.server_id = $1
+  AND databases.hidden = $2
+ORDER BY databases.id + $zero ASC  -- Added +0 to prevent index sort usage
+LIMIT $3;`}
+</CodeBlock>
+
+### The `+ $zero` technique
+This Postgres optimization technique:
+- **Prevents sort optimization**: Postgres can't use the primary key index for sorting
+- **Forces filter-first strategy**: Must use a better index for efficient filtering of server_id and hidden columns
+- **Maintains correctness**: Adding zero doesn't change sort order
+- **Triggers replanning**: Forces Postgres to reconsider index selection strategy
+
+This is a well-known Postgres expert technique that Query Advisor automates for you.
+
+## Step 5: Benchmark the optimization
+
+We executed the modified query using the collector workflow:
+
+### Performance improvement
+- **Baseline runtime**: 495.51ms → **58.11ms** (8.5x improvement)
+- **Cost estimate**: 39,259 → 60,680 (higher estimated cost)
+- **Plan strategy**: Changed from Index Scan to Sort + Bitmap Heap Scan approach
+- **Index usage**: Switched to index_databases_on_server_id_and_datname
+
+### Plan comparison
+The workbook's compare view highlighted the key changes:
+- **Plan A (Baseline)**: Gather Merge → Index Scan on databases_pkey
+- **Plan B (Variant 1)**: Limit → Sort → Bitmap Heap Scan → Bitmap Index Scan
+- **Runtime improvement**: 495.51ms → 58.11ms
+- **Index strategy**: Primary key scan → server_id + datname index
+
+![Workbooks plan comparison showing performance improvements between baseline and optimized query variants](workbooks-compare-plans.png)
+
+## Step 6: Validate the results
+
+The dramatic improvement confirmed Query Advisor's analysis was correct:
```

Comment:
Uhm, this sentence is strange - or maybe sentence itself is okay, but it can end with "." instead of ":"